### ISSUE TITLE
trouble with `eigs`

### ISSUE THREAD

**Issue by hrothgar** [chebfun/chebfun#1020]
Here is Davies’ complex harmonic oscillator:
```
% Davies complex harmonic oscillator

% Domain, operator, boundary conditions
dom = [-8, 8];
x   = chebfun('x', dom);
op  = @(x,u) -diff(u,2) + (1i*x.^2).*u;
bc  = @(x,u) [ u(-8)
               u(8) ];

% Build the chebop
A = chebop(op, dom, bc);

% Compute eigenpairs
[ev, ew] = eigs(A, 7);
ev = quasimatrix(ev);
```
I can see that the residual is small:
```
norm(A*ev - ew*ev)
```
```
ans =
     1.684460424149466e-09
```
If I instead compute 8 or 9 eigenpairs, the accuracy is killed (which is what we expect, because the higher eigenvalues are exponentially sensitive to perturbations in the operator):
```
norm(A*ev - ew*ev)
```
```
ans =
  80.162350842885928
```
But if I try for 10 or more eigenpairs, I actually get an error upon computing the residual norm:
```
norm(A*ev - ew*ev)
```
```
Index exceeds matrix dimensions.
Error in chebtech/times>coeff_times (line 167)
hc = hc(1:mn,:);                                % Take the first half.
Error in chebtech/times>coeff_times_main (line 133)
    coeffs = coeff_times( f, g );
Error in chebtech/times (line 69)
[f.coeffs, pos] = coeff_times_main(f.coeffs, g.coeffs);
Error in classicfun/times (line 35)
    f.onefun = times(f.onefun, g.onefun, varargin{:});
Error in chebfun/times (line 90)
            f.funs{k} = times(f.funs{k}, g.funs{k});
Error in chebfun/power>columnPower (line 154)
        g = f.*f;
Error in chebfun/power (line 70)
            g = columnPower(f, b, pref);
Error in @(x,u)-diff(u,2)+(1i*x.^2).*u

Error in chebop/feval (line 185)
    out = N.op(x, u);
Error in chebop/mtimes (line 25)
    C = feval(A, B);
```
In this last case, the error comes from the product `A*ev`.

The eigenfunctions appear to be the same in every case, so part of it might be an ordering issue (?).

**Comment by aaustin141** [comment]
The example originally posted here as working now crashes for me with a dimension-mismatch error:
```
>> dom = [-8, 8];
>> x   = chebfun('x', dom);
>> op  = @(x,u) -diff(u,2) + (1i*x.^2).*u;
>> bc  = @(x,u) [ u(-8)
                  u(8) ];
>> A = chebop(op, dom, bc);
>> [ev, ew] = eigs(A, 7);
>> ev = quasimatrix(ev);
>> norm(A*ev - ew*ev)
Error using * (line 52)
Matrix dimensions must agree.
Error in * (line 84)
        f = mtimes(g.', f.').'; 
```

**Comment by asgeirbirkis** [comment]
Closed by #1579.
